File: Seccion 17 - Archivos o Ficheros en C++/Seccion_17_Python/02_ejercicio_01.py

def escribir_frase():
    nombre_archivo: str
    frase: str
    rpt: str = 's'
    
    try:
        nombre_archivo = input("\nDigite el nombre del archivo: ")
        archivo = open(nombre_archivo, "w")
    except FileNotFoundError:
        print("No se pudo abrir el fichero.")
        exit()

    while rpt == 'S' or rpt == 's':
        frase = input("\nDigite una frase: ")
        archivo.write(frase + "\n")
        
        rpt = input("\nDesea escribir otra frase? (s/n): ")
    
    archivo.close()

def main():
    escribir_frase()

if __name__ == "__main__":
    main()
        

File: Seccion 17 - Archivos o Ficheros en C++/02_ejercicio_01.cpp
/*
    01. Realice un programa que pida al usuario el nombre de un fichero
    de texto y, a continuación permita almacenar al usuario tantas frases
    como el mismo desee.
*/

#include <iostream>
#include <stdlib.h>
#include <fstream>

using namespace std;

void escribirFrase();

int main() {

    escribirFrase();

    system("pause");
    return 0;
}

void escribirFrase() {
    ofstream archivo;
    string nombreArchivo, frase;
    char rpt;

    cout << "\nDigite el nombre del archivo: ";
    getline(cin, nombreArchivo);

    archivo.open(nombreArchivo.c_str(), ios::out); // Abriendo el archivo

    if (archivo.fail()) {
        cout << "No se pudo abrir el archivo." << endl;
        exit(1);
    }

    do {
        cin.ignore();
        cout << "\nDigite una frase: ";
        getline(cin, frase);
        archivo << frase << endl;

        cout << "\nDesea agregar otra frase? (s/n): ";
        cin >> rpt;
    } while((rpt == 's') || (rpt == 'S'));

    archivo.close(); // Cerramos el archivo
}